core/trino-main/src/main/java/io/trino/sql/planner/LocalExecutionPlanner.java

```diff
@@ -1621,14 +1621,16 @@ public PhysicalOperation visitTopN(TopNNode node, LocalExecutionPlanContext cont
                 sortOrders.add(node.getOrderingScheme().getOrdering(symbol));
             }
 
+            DataSize maxPartialTopNMemorySize = SystemSessionProperties.getMaxPartialTopNMemory(session);
             OperatorFactory operator = TopNOperator.createOperatorFactory(
                     context.getNextOperatorId(),
                     node.getId(),
                     source.getTypes(),
                     (int) node.getCount(),
                     sortChannels,
                     sortOrders,
-                    plannerContext.getTypeOperators());
+                    plannerContext.getTypeOperators(),
+                    node.getStep() != TopNNode.Step.PARTIAL || maxPartialTopNMemorySize.compareTo(DataSize.ofBytes(0)) <= 0 ? Optional.empty() : Optional.of(maxPartialTopNMemorySize));
```

Comment:
I think this would be easier to read if we flipped the ternary condition:

```java
node.getStep() == TopNNode.Step.PARTIAL && maxPartialTopNMemorySize.compareTo(DataSize.ofBytes(0)) > 0 : Optional.of(maxPartialTopNMemorySize)) ? Optional.empty();
```

Reply:
Done.